pkg/configprofiles/profiles.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package configprofiles
+
+import (
+	"sort"
+
+	"github.com/cockroachdb/cockroach/pkg/clusterversion"
+	"github.com/cockroachdb/cockroach/pkg/server/autoconfig/autoconfigpb"
+)
+
+var staticProfileNames = func() (res []string) {
+	for name := range staticProfiles {
+		res = append(res, name)
+	}
+	sort.Strings(res)
+	return res
+}()
+
+const defaultProfileName = "default"
+
+var staticProfiles = map[string][]autoconfigpb.Task{
+	defaultProfileName: {
+		// Do not define tasks in the "default" profile. It should continue
+		// to generate clusters with default configurations.
+	},
+	// The "example" profile exists for demonstration and documentation
+	// purposes.
+	"example": {
+		makeTask("create an example database",
+			/* nonTxnSQL */ []string{
+				"CREATE DATABASE IF NOT EXISTS example",
+				"CREATE TABLE IF NOT EXISTS example.data AS SELECT 'hello' AS value",
+			},
+			nil, /* txnSQL */
+		),
+	},
+	"multitenant+shared+noapp": multitenantClusterInitTasks,
+	"multitenant+shared+app": append(
+		multitenantClusterInitTasks,
+		makeTask("create an application tenant",
+			nil, /* nonTxnSQL */
+			/* txnSQL */ []string{
+				// Create the app tenant record.
+				"CREATE TENANT application",
+				// Run the service for the application tenant.
+				"ALTER TENANT application START SERVICE SHARED",
+			},
+		),
+		makeTask("activate application tenant",
+			/* nonTxnSQL */ []string{
+				// Make the app tenant receive SQL connections by default.
+				"SET CLUSTER SETTING server.controller.default_tenant = 'application'",
+			},
+			nil, /* txnSQL */
+		),
+	),
+}
+
+var multitenantClusterInitTasks = []autoconfigpb.Task{
+	makeTask("initial cluster config",
+		/* nonTxnSQL */ []string{
+			// Disable RU limits.
+			"SET CLUSTER SETTING kv.tenant_rate_limiter.burst_limit_seconds = 10000000",
+			"SET CLUSTER SETTING kv.tenant_rate_limiter.rate_limit = -1000",
+			"SET CLUSTER SETTING kv.tenant_rate_limiter.read_batch_cost = 0",
+			"SET CLUSTER SETTING kv.tenant_rate_limiter.read_cost_per_mebibyte = 0",
+			"SET CLUSTER SETTING kv.tenant_rate_limiter.write_cost_per_megabyte = 0",
+			"SET CLUSTER SETTING kv.tenant_rate_limiter.write_request_cost = 0",
```

Comment:
It's unclear if we'll need this now that this rate limiter is attached to the capability system.